### PR TITLE
Randomize more settings

### PR DESCRIPTION
**Pull request by CurtizJ**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by CurtizJ** [comment]
#39667.

File: tests/clickhouse-test [hunk 1]
```diff
@@ -395,6 +395,16 @@ class FailureReason(enum.Enum):
     INTERNAL_ERROR = "Test internal error: "
 
 
+def default_generator_for_bytes_setting():
+    return (
+        lambda: 0
+        if random.random() < 0.5
+        else 1
```
Comment (nickitat) [inline tests/clickhouse-test:402]: i'm afraid that it may slow down some tests significantly

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
@CurtizJ, this is a so good pull request; let's continue!

**Comment by alexey-milovidov** [comment]
We have faced a bug on GitHub.
GitHub can reset file mode, dropping the executable flag.

**Comment by alexey-milovidov** [comment]
@CurtizJ This is so wonderful, let's continue!

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit fe2f267dbe6202ec757904b061460142233c9f7b with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/39663/fe2f267dbe6202ec757904b061460142233c9f7b/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by alexey-milovidov** [comment]
We need this.

**Comment by alexey-milovidov** [comment]
@CurtizJ, We still need this.

**Comment by alexey-milovidov** [comment]
We need this.

**Comment by alexey-milovidov** [comment]
@CurtizJ We need this PR to be merged.

**Comment by alexey-milovidov** [comment]
@CurtizJ This PR makes sense.

**Comment by alexey-milovidov** [comment]
How nice it is.

**Comment by alexey-milovidov** [comment]
This pull request is good to continue.

**Comment by alexey-milovidov** [comment]
This is a good pull request - let's continue.

**Comment by alexey-milovidov** [comment]
We need this pull request to be merged.

**Comment by alexey-milovidov** [comment]
@CurtizJ, please continue :smile_cat:

**Comment by alexey-milovidov** [comment]
Hi there. We need to continue.

**Comment by alexey-milovidov** [comment]
@CurtizJ, we want this feature so much!

**Comment by CurtizJ** [comment]
Current (not complete) list of failed tests:
```
02480_max_map_null_totals
02151_hash_table_sizes_stats
01244_optimize_distributed_group_by_sharding_key
00155_long_merges
01064_incremental_streaming_from_2_src_with_feedback
01514_distributed_cancel_query_on_error
02896_memory_accounting_for_user
02352_rwlock
01913_exact_rows_before_limit_full
02178_column_function_insert_from
01134_max_rows_to_group_by
00184_shard_distributed_group_by_no_merge
01710_projections_partial_optimize_aggregation_in_order
01563_distributed_query_finish
01710_aggregate_projections
01088_array_slice_of_aggregate_functions
00166_functions_of_aggregation_states
01012_reset_running_accumulate
02496_remove_redundant_sorting
01513_optimize_aggregation_in_order_memory_long
02500_remove_redundant_distinct
02500_remove_redundant_distinct_analyzer
02496_remove_redundant_sorting_analyzer
01472_many_rows_in_totals
01591_window_functions
01563_distributed_query_finish
02366_kql_extend
02352_rwlock
02366_kql_makeseries
01799_long_uniq_theta_sketch
02427_column_nullable_ubsan
02479_race_condition_between_insert_and_droppin_mv	
```

**Comment by CurtizJ** [comment]
`01710_projections_partial_optimize_aggregation_in_order` - #57692

**Comment by CurtizJ** [comment]
`01291_distributed_low_cardinality_memory_efficient` - #57814.

**Comment by CurtizJ** [comment]
So I found 4 real issues in this PR: #57686, #57688, #57692, #57814

Two of them (#57686 and #57692) are fixed in #57795 and #57790.

#57688 is closed with `won't fix` resolution because it is fixed with new analyzer.

#57814 looks a bit weird. Probably should be closed with `won't fix` unless the more real scenario will be provided.

**Comment by CurtizJ** [comment]
Last failing test:
```
02274_full_sort_join_nodistinct

2023-12-14 09:58:47 [6b3bb7ac6f8c] 2023.12.14 09:58:47.700559 [ 2600 ] {c955c4e1-82d4-474e-bec2-77bdd0cbbeff} <Error> executeQuery: std::exception. Code: 1001, type: std::__1::__fs::filesystem::filesystem_error, e.what() = filesystem error: in create_directories: No such file or directory ["/var/lib/clickhouse/caches/cached_s3/969/969b1806ff4241db78e954e0cd3840c0"]
2023-12-14 09:58:47 Cannot print extra info for Poco::Exception (version 23.12.1.594) (from [::1]:50854) (comment: 02274_full_sort_join_nodistinct.gen.sql) (in query: SELECT t1.key, t2.key, t1.s, length(t2.s) FROM tn1 AS t1 ALL LEFT JOIN tn2 AS t2 ON t1.key == t2.key ORDER BY t1.key, t2.key, t1.s;), Stack trace (when copying this message, always include the lines below):
2023-12-14 09:58:47 
2023-12-14 09:58:47 0. ./contrib/llvm-project/libcxx/include/string:1499: std::system_error::system_error(std::error_code, String const&) @ 0x0000000016ad8615 in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 1. ./contrib/llvm-project/libcxx/include/__filesystem/filesystem_error.h:42: std::__fs::filesystem::filesystem_error::filesystem_error[abi:v15000](String const&, std::__fs::filesystem::path const&, std::error_code) @ 0x000000000c597901 in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 2. ./contrib/llvm-project/libcxx/include/__filesystem/filesystem_error.h:90: void std::__fs::filesystem::__throw_filesystem_error[abi:v15000]<String&, std::__fs::filesystem::path const&, std::error_code const&>(String&, std::__fs::filesystem::path const&, std::error_code const&) @ 0x0000000016a8f01a in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 3. ./contrib/llvm-project/libcxx/src/filesystem/filesystem_common.h:0: std::__fs::filesystem::detail::(anonymous namespace)::ErrorHandler<bool>::report(std::error_code const&) const (.llvm.14483552974555787744) @ 0x0000000016a92376 in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 4. ./build_docker/./contrib/llvm-project/libcxx/src/filesystem/operations.cpp:0: std::__fs::filesystem::__create_directories(std::__fs::filesystem::path const&, std::error_code*) @ 0x0000000016a92b79 in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 5. ./contrib/llvm-project/libcxx/include/string:1499: DB::TemporaryDataOnDisk::createCacheFile(unsigned long) @ 0x00000000112ef13a in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 6. ./build_docker/./src/Interpreters/TemporaryDataOnDisk.cpp:83: DB::TemporaryDataOnDisk::createStream(DB::Block const&, unsigned long) @ 0x00000000112ef572 in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 7. ./build_docker/./src/Processors/Transforms/MergeSortingTransform.cpp:171: DB::MergeSortingTransform::consume(DB::Chunk) @ 0x00000000125e3e09 in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 8. ./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:701: DB::SortingTransform::work() @ 0x00000000125fa444 in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 9. ./build_docker/./src/Processors/Executors/ExecutionThreadContext.cpp:0: DB::ExecutionThreadContext::executeTask() @ 0x000000001235335a in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 10. ./build_docker/./src/Processors/Executors/PipelineExecutor.cpp:273: DB::PipelineExecutor::executeStepImpl(unsigned long, std::atomic<bool>*) @ 0x000000001234a070 in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 11. ./contrib/llvm-project/libcxx/include/__memory/shared_ptr.h:833: void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<DB::PipelineExecutor::spawnThreads()::$_0, void ()>>(std::__function::__policy_storage const*) @ 0x000000001234b0ef in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 12. ./base/base/../base/wide_integer_impl.h:809: ThreadPoolImpl<ThreadFromGlobalPoolImpl<false>>::worker(std::__list_iterator<ThreadFromGlobalPoolImpl<false>, void*>) @ 0x000000000c631670 in /usr/lib/debug/usr/bin/clickhouse.debug
2023-12-14 09:58:47 13. ./build_docker/./src/Common/ThreadPool.cpp:0: void std::
```

**Comment by CurtizJ** [comment]
`02274_full_sort_join_nodistinct` is fixed in #58013.

**Comment by CurtizJ** [comment]
Tests look ok except `00155_long_merges` in flaky check, but I think it won't pass even on master with parallel run.

**Comment by CurtizJ** [comment]
Anyway there will some new flaky tests.

**Pull request merged**